src/ty/composite.rs

```diff
@@ -91,7 +91,7 @@ impl IntoPortable for TypeDefComposite {
 
 impl TypeDefComposite {
     /// Creates a new struct definition with named fields.
-    pub fn new<I>(fields: I) -> Self
+    pub(crate) fn new<I>(fields: I) -> Self
```

Comment:
Why this change?

Reply:
So users can't instantiate with any fields containing a `PhantomData`. They are filtered out in the builder, and this would allow circumventing the builder. 

The other option of course would be to do the phantom filtering directly in this method, but I think we should encourage the use of the builder as the way to construct `Composite` types.